statistics/handle/update.go

```diff
@@ -1230,14 +1230,11 @@ func (h *Handle) autoAnalyzePartitionTableInDynamicMode(tblInfo *model.TableInfo
 		sqlBuilder.WriteString(suffix)
 		return sqlBuilder.String()
 	}
-	if len(partitionNames) < 1 {
```

Comment:
Why do we remove this?

Reply:
When the if branch exists, we would not go to the part of checking index stats(old Line1258-1289).